api/controllers/AuthController.ts

```diff
@@ -65,6 +71,15 @@ export class AuthController {
     return { error: null };
   }
 
+  @UseBefore(UserAuthentication)
+  @Post('/modifyEmail')
+  async modifyEmail(@Body() modifyEmailRequest: ModifyEmailRequest,
+    @AuthenticatedUser() user: UserModel): Promise<ModifyEmailResponse> {
+    await this.userAuthService.modifyEmail(user, modifyEmailRequest.email);
+    await this.emailService.sendEmailVerification(user.email, user.firstName, user.accessCode);
+    return { error: null };
```

Comment:
nit: can we set the user's email as `toLowerCased`? (see `AuthController::resendEmailVerification()` for implementation). We've had issues in the past where a user accidentally capitalizes a character in their email, and thus wouldn't receive an email from us to that account since it doesn't exist, when in reality it was meant to be lowercased (basically, this allows users to typo the case)